### PR TITLE
Lsdd updates

### PR DESCRIPTION
**Pull request by ojcobb**
A few bug fixes:

- Ensure when updating reference data it gets normalised if necessary. (#261)
- Catch overflow/underflow caused by too high dimensionality.
- Catch when error is caused by too many repeat samples. (And don't test using repeat samples)

### CODE REVIEW

File: alibi_detect/utils/pytorch/distance.py [hunk 1]
```diff
@@ -120,6 +120,13 @@ def permed_lsdds(
     Vector of B LSDD estimates for each permutation, H_lam_inv which may have been inferred, and optionally
     the unpermed LSDD estimate.
     """
+
+    # Check for overflow/underflow caused by too high dimensionality
+    if (~torch.isfinite(H)).any():
+        raise ValueError(
+            "Overflow or underflow occured. Trying reducing dimensionality or trying "
```
Comment (arnaudvl) [inline alibi_detect/utils/pytorch/distance.py:127]: small typo -> "try" instead of trying 2x

File: alibi_detect/utils/tensorflow/distance.py [hunk 2]
```diff
@@ -149,6 +149,13 @@ def permed_lsdds(
     Vector of B LSDD estimates for each permutation, H_lam_inv which may have been inferred, and optionally
     the unpermed LSDD estimate.
     """
+
+    # Check for overflow/underflow caused by too high dimensionality
+    if tf.reduce_any(~tf.math.is_finite(H)):
+        raise ValueError(
+            "Overflow or underflow occured. Trying reducing dimensionality or trying "
```
Comment (arnaudvl) [inline alibi_detect/utils/tensorflow/distance.py:156]: Also try instead of trying

File: alibi_detect/utils/pytorch/distance.py (hunk 1 above)
Comment (ojcobb) [reply alibi_detect/utils/pytorch/distance.py]: Ah yeah thanks. Fixed.
